runtimes/pythonrt/runner/values.py

```diff
@@ -0,0 +1,127 @@
+        if isinstance(v, list):
+            return pb.Value(list=pb.List(vs=[self.wrap(x) for x in v]))
+        if isinstance(v, set):
+            return pb.Value(set=pb.Set(vs=[self.wrap(x) for x in v]))
+        if isinstance(v, dict):
+            return pb.Value(
+                dict=pb.Dict(
+                    items=[pb.Dict.Item(k=self.wrap(k), v=self.wrap(v)) for k, v in v.items()]
+                )
+            )
+        if isinstance(v, bytes):
+            return pb.Value(bytes=pb.Bytes(v=v))
+        if isinstance(v, datetime):
+            v = v.astimezone(UTC)
+            ts = Timestamp()
+            ts.FromDatetime(v)
+            return pb.Value(time=pb.Time(v=ts))
+        if isinstance(v, timedelta):
+            d = Duration()
+            d.FromTimedelta(v)
+            return pb.Value(duration=pb.Duration(v=d))
+
+        if hasattr(v, "__dict__"):
+            return pb.Value(
+                struct=pb.Struct(
+                    ctor=self.wrap(v.__class__.__name__),
+                    fields={k: self.wrap(v) for k, v in v.__dict__.items()},
+                )
+            )
+
+        if hasattr(v, "__slots__"):
+            return pb.Value(
+                struct=pb.Struct(
+                    ctor=self.wrap(v.__class__.__name__),
+                    fields={k: self.wrap(getattr(v, k)) for k in v.__slots__},
+                )
+            )
+
+        raise ValueError(f"unsupported type: {type(v)}")
+
+    # TODO: This is an instance method as we might add options to the wrapper in the future.
+    def unwrap(self, v: pb.Value) -> any:
+        """Unwrap an autokitteh value into a python value.
+
+        Note that wrap and unwrap are guaranteed to be symmetric.
+        Two notable examples:
+            unwrap(wrap((1, 2))) = [1, 2]
+
+            class C:
+                def __init__(self):
+                    self.x = 42
+
+            unwrap(C()) = namedtuple("C", {"x"})(42)
+        """
+
+        if v.HasField("nothing"):
+            return None
+        if v.HasField("integer"):
+            return v.integer.v
+        if v.HasField("float"):
+            return v.float.v
+        if v.HasField("string"):
+            return v.string.v
+        if v.HasField("boolean"):
+            return v.boolean.v
+        if v.HasField("list"):
+            return [self.unwrap(x) for x in v.list.vs]
+        if v.HasField("set"):
+            return set([self.unwrap(x) for x in v.set.vs])
+        if v.HasField("dict"):
+            return {self.unwrap(x.k): self.unwrap(x.v) for x in v.dict.items}
+        if v.HasField("bytes"):
+            return v.bytes.v
+        if v.HasField("struct"):
+            tpl = namedtuple(str(self.unwrap(v.struct.ctor)), v.struct.fields.keys())
+            return tpl(*[self.unwrap(x) for x in v.struct.fields.values()])
+        if v.HasField("time"):
+            return datetime.fromtimestamp(v.time.v.seconds, UTC)
```

Comment:
You are losing precision. Use 

```suggestion
            return v.time.v.ToDatetime(UTC)
```

Reply:
done